### DISCUSSION TITLE
Is lookup strict?

### DISCUSSION THREAD

**Discussion by cbizon** [NCATSTranslator/TranslatorArchitecture#65] [General]
When a TRAPI query comes into an ARA or KP, what is that tool allowed to do?  

Some tools treat this as a database lookup - you have it or you don't, and you return exactly what is asked for.

Other tools consider the query a suggestion and expand or conflate entities, combine results, etc.

Is this flexibility desired?  Can we manage it, perhaps using operations or query options?

**Comment by remontoire-pac** [comment]
I would argue that the better KPs get, the better they will be at
expanding, conflating, combining in advance, so yes, I would say this
flexibility is desired. But it needs to be transparent, too, since it
expands our EPC problem:

*Source X0 found query Q0 with Y0 attributes, but also source X1 found Q1
by conflation mapping C1 using element E1, and source X2 found Q2 by
similarity to Q0 using similarity measure S2.*
